client/src/protocol.ts

```diff
@@ -64,6 +64,14 @@ export class GameSocket extends EventEmitter {
 
 export class Entity {
     constructor(public id: number, public position: Victor, public direction?: Victor) { }
+
+    public distanceSq(to: Entity | Victor): number {
+        if ((<any>to)['position']) {
```

Comment:
You might consider changing this to `'position' in to` to be just a little more clear, but this is fine.